app/backends/workflow-orchestration.ts

```diff
@@ -158,25 +186,29 @@ export async function updateWorkItem(req: HarmonyRequest, res: Response): Promis
         workItem.workflowStepIndex + 1,
       );
 
+      const successWorkItemCount = await workItemCountForStep(
+        tx,
+        workItem.jobID,
+        workItem.workflowStepIndex,
+        WorkItemStatus.SUCCESSFUL,
```

Comment:
Another now or later thing, we have a ticket that arrived in the backlog for allowing a "success with warnings". It makes me think that this should really be some sort of function that returns a list of states that constitute success, or it should be all wrapped up in a `successfulWorkItemCount` function or something so we can more freely change it.

You don't need to do it as part of this PR, but it may be an overall time savings to write the function now vs trying to find that we need it here later.

Reply:
I changed  the optional `status` parameter of `workItemCountForStep` to take either a string or an array of strings to allow us to have multiple 'successful' `WorkItemStatus`s. So for now we pass it `[WorkItemStatus.SUCCESSFUL]`, but later on we might pass it `[WorkItemStatus.SUCCESSFUL, WorkItemStatus.SUCCESSFUL_WITH_WARNINGS]`.